lib/reek/smells/subclassed_from_core_class.rb

```diff
@@ -6,26 +6,49 @@ module Reek
   module Smells
     # TODO: documentation
     class SubclassedFromCoreClass < SmellDetector
-      CORE_CLASSES = [:Array, :Hash, :String]
+
+      METHODS = {
+        casgn: :inspect_casgn,
+        class: :inspect_class,
+      }.freeze
 
       def self.contexts
-        [:class]
+        [:class, :casgn]
       end
 
       # Checks +ctx+ for if it is subclasssed from a core class
       #
       # @return [Array<SmellWarning>]
       def inspect(ctx)
-        _class_node, ancestor_node, _body_node = ctx.exp.to_a
-        return [] if ancestor_node.nil?
-        ancestor_namespace, ancestor = ancestor_node.to_a
-        return [] unless CORE_CLASSES.include?(ancestor) && ancestor_namespace.nil?
+        public_send(METHODS[ctx.type], ctx)
+      end
+
+      def inspect_class(ctx)
+        expression = ctx.exp
+        superclass = expression.superclass
+
+        return [] unless expression.superclass? && superclass.core_class?
+
+        [smell_warning(
+          context: ctx,
+          lines: [ctx.exp.line],
+          message: "inherits from a core class #{superclass}",
+          parameters: { ancestor: superclass.name })]
+      end
+
+      def inspect_casgn(ctx)
+        expression = ctx.exp
+        assignment = expression.assignment
+
+        return [] unless expression.class_creation? && assignment.emitted.core_class?
```

Comment:
Since `expression` has a method `class_creation?`, you can also add a method `superclass`, which will allow you to unify `inspect_casgn` and `inspect_class`.